Rpackage Version 1.0.0.2 crashes RStudio

Running the R package Version 1.0.0.2 crashes RStudio. This occurs  at the xgb.DMatrix call in the xgbtrain example. As well as when using saved xgboost objects fit under v0.90.0.2 or fitting new models.

data(agaricus.train, package='xgboost')
data(agaricus.test, package='xgboost')
dtrain <- xgb.DMatrix(agaricus.train$data, label = agaricus.train$label)


R.version
platform       x86_64-apple-darwin15.6.0   
arch           x86_64                      
os             darwin15.6.0                
system         x86_64, darwin15.6.0        
status                                     
major          3                           
minor          5.2                         
year           2018                        
month          12                          
day            20                          
svn rev        75870                       
language       R                           
version.string R version 3.5.2 (2018-12-20)
nickname       Eggshell Igloo 

RStudio Version 1.2.5033

Here is a minimal code example, if it helps:
```R
library(xgboost)
xgb.DMatrix(matrix(1:2, nrow=1))
```
The R interpreter gives the following error mesages and quits:
```
OMP: Error #15: Initializing libomp.dylib, but found libomp.dylib already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://openmp.llvm.org/
zsh: abort      R
```
The package xgboost was installed from CRAN and compiled with clang:
```bash
clang --version
Apple clang version 11.0.3 (clang-1103.0.32.59)
Target: x86_64-apple-darwin19.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
Full install log under details below.
I have also a real g++ compiler installed via homebrew.
```g++-9 --version
g++-9 (Homebrew GCC 9.2.0_3) 9.2.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
As opposed to the g++ of clang (on macOS):
```
g++ --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.3 (clang-1103.0.32.59)
Target: x86_64-apple-darwin19.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

<details>

```R
> install.packages("xgboost")

  There is a binary version available but the source version is later:
          binary  source needs_compilation
xgboost 0.90.0.2 1.0.0.2              TRUE

Do you want to install from sources the package which needs compilation? (Yes/no/cancel) Yes
installing the source package ‘xgboost’

trying URL 'https://cran.rstudio.com/src/contrib/xgboost_1.0.0.2.tar.gz'
Content type 'application/x-gzip' length 898104 bytes (877 KB)
==================================================
downloaded 877 KB

* installing *source* package ‘xgboost’ ...
** package ‘xgboost’ successfully unpacked and MD5 sums checked
** using staged installation
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking Backtrace lib... 
checking for backtrace in -lexecinfo... no
checking endian... 
checking whether OpenMP will work in a package... yes
configure: creating ./config.status
config.status: creating src/Makevars
** libs
Makevars:17: -DXGBOOST_STRICT_R_MODE=1
Makevars:17: -DDMLC_LOG_BEFORE_THROW=0
Makevars:17: -DDMLC_ENABLE_STD_THREAD=1
Makevars:17: -DDMLC_DISABLE_STDIN=1
Makevars:17: -DDMLC_LOG_CUSTOMIZE=1
Makevars:17: -DXGBOOST_CUSTOMIZE_LOGGER=1
Makevars:17: -DRABIT_CUSTOMIZE_MSG_
Makevars:17: -DRABIT_STRICT_CXX98_
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -Xclang -fopenmp -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -fPIC  -Wall -g -O2  -c xgboost_R.cc -o xgboost_R.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -Xclang -fopenmp -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -fPIC  -Wall -g -O2  -c xgboost_custom.cc -o xgboost_custom.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c xgboost_assert.c -o xgboost_assert.o
clang -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c init.c -o init.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -Xclang -fopenmp -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -fPIC  -Wall -g -O2  -c amalgamation/xgboost-all0.cc -o amalgamation/xgboost-all0.o
In file included from amalgamation/xgboost-all0.cc:33:
In file included from amalgamation/../src/data/data.cc:13:
In file included from amalgamation/../src/data/sparse_page_writer.h:20:
./dmlc-core/include/dmlc/concurrency.h:36:22: warning: braces around scalar initializer [-Wbraced-scalar-init]
  Spinlock() : lock_(ATOMIC_FLAG_INIT) {
                     ^~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/atomic:2437:26: note: expanded from macro 'ATOMIC_FLAG_INIT'
#define ATOMIC_FLAG_INIT {false}
                         ^~~~~~~
In file included from amalgamation/xgboost-all0.cc:38:
amalgamation/../src/data/ellpack_page_source.cc:33:11: warning: variable 'page' is uninitialized when used here [-Wuninitialized]
  return *page;
          ^~~~
amalgamation/../src/data/ellpack_page_source.cc:32:20: note: initialize the variable 'page' to silence this warning
  EllpackPage* page;
                   ^
                    = nullptr
amalgamation/../src/data/ellpack_page_source.cc:40:11: warning: variable 'page' is uninitialized when used here [-Wuninitialized]
  return *page;
          ^~~~
amalgamation/../src/data/ellpack_page_source.cc:39:20: note: initialize the variable 'page' to silence this warning
  EllpackPage* page;
                   ^
                    = nullptr
In file included from amalgamation/xgboost-all0.cc:75:
amalgamation/../src/common/io.cc:102:8: warning: unused variable 'ReadErr' [-Wunused-variable]
  auto ReadErr = [&fname]() {
       ^
In file included from amalgamation/xgboost-all0.cc:13:
In file included from amalgamation/../src/metric/metric.cc:6:
In file included from ./dmlc-core/include/dmlc/./registry.h:9:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/map:479:
In file included from /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__tree:15:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:2338:5: warning: delete called on non-final 'xgboost::tree::QuantileHistMaker::Builder' that has virtual functions but non-virtual destructor [-Wdelete-non-abstract-non-virtual-dtor]
    delete __ptr;
    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:2651:7: note: in instantiation of member function 'std::__1::default_delete<xgboost::tree::QuantileHistMaker::Builder>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
amalgamation/../src/tree/updater_quantile_hist.cc:74:14: note: in instantiation of member function 'std::__1::unique_ptr<xgboost::tree::QuantileHistMaker::Builder, std::__1::default_delete<xgboost::tree::QuantileHistMaker::Builder> >::reset' requested here
    builder_.reset(new Builder(
             ^
5 warnings generated.
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -Xclang -fopenmp -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -fPIC  -Wall -g -O2  -c amalgamation/dmlc-minimum0.cc -o amalgamation/dmlc-minimum0.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -Xclang -fopenmp -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -fPIC  -Wall -g -O2  -c rabit/src/engine_empty.cc -o rabit/src/engine_empty.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./include -I./dmlc-core/include -I./rabit/include -I. -DXGBOOST_STRICT_R_MODE=1 -DDMLC_LOG_BEFORE_THROW=0 -DDMLC_ENABLE_STD_THREAD=1 -DDMLC_DISABLE_STDIN=1 -DDMLC_LOG_CUSTOMIZE=1 -DXGBOOST_CUSTOMIZE_LOGGER=1 -DRABIT_CUSTOMIZE_MSG_ -DRABIT_STRICT_CXX98_  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -Xclang -fopenmp -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -fPIC  -Wall -g -O2  -c rabit/src/c_api.cc -o rabit/src/c_api.o
clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o xgboost.so ./xgboost_R.o ./xgboost_custom.o ./xgboost_assert.o ./init.o ./amalgamation/xgboost-all0.o ./amalgamation/dmlc-minimum0.o ./rabit/src/engine_empty.o ./rabit/src/c_api.o -Xclang -fopenmp /usr/local/lib/libomp.dylib -DDMLC_CMAKE_LITTLE_ENDIAN=1 -pthread -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Library/Frameworks/R.framework/Versions/3.6/Resources/library/00LOCK-xgboost/00new/xgboost/libs
** R
** data
** demo
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (xgboost)
```
</details>

--- a/R-package/configure
+++ b/R-package/configure
@@ -2710,7 +2710,7 @@ fi
 if test `uname -s` = "Darwin"
 then
   OPENMP_CXXFLAGS='-Xclang -fopenmp'
-  OPENMP_LIB='/usr/local/lib/libomp.dylib'
+  OPENMP_LIB='-lomp'
   ac_pkg_openmp=no
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether OpenMP will work in a package" >&5
 $as_echo_n "checking whether OpenMP will work in a package... " >&6; }

--- a/R-package/configure.ac
+++ b/R-package/configure.ac
@@ -29,7 +29,7 @@ fi
 if test `uname -s` = "Darwin"
 then
   OPENMP_CXXFLAGS='-Xclang -fopenmp'
-  OPENMP_LIB='/usr/local/lib/libomp.dylib'
+  OPENMP_LIB='-lomp'
   ac_pkg_openmp=no
   AC_MSG_CHECKING([whether OpenMP will work in a package])
   AC_LANG_CONFTEST([AC_LANG_PROGRAM([[#include <omp.h>]], [[ return (omp_get_max_threads() <= 1); ]])])

